packages/account/src/providers/transaction-summary/input.ts

```diff
@@ -33,16 +33,32 @@ export function getInputsContract(inputs: Input[]) {
 }
 
 /** @hidden */
-export function getInputFromAssetId(inputs: Input[], assetId: string) {
+function findCoinInput(inputs: Input[], assetId: string): InputCoin | undefined {
   const coinInputs = getInputsCoin(inputs);
-  const messageInputs = getInputsMessage(inputs);
-  const coinInput = coinInputs.find((i) => i.assetId === assetId);
-  // TODO: should include assetId in InputMessage as well. for now we're mocking ETH
-  const messageInput = messageInputs.find(
-    (_) => assetId === '0x0000000000000000000000000000000000000000000000000000000000000000'
-  );
-
-  return coinInput || messageInput;
+  return coinInputs.find((i) => i.assetId === assetId);
+}
+
+/** @hidden */
+function findMessageInput(inputs: Input[]): InputMessage | undefined {
+  return getInputsMessage(inputs)?.[0];
+}
+/** @hidden */
+export function getInputFromAssetId(
+  inputs: Input[],
+  assetId: string,
+  isBaseAsset = false
```

Comment:
make sense to receive `baseAssetId` , and do the comparison `assetId === baseAssetId` inside of this method?  

would be more smooth as normally the operations already have the `baseAssetId` avoiding the dev needing to provide `assetId === baseAssetId` everytime using this method

Reply:
This method is used in [`getWithdrawFromFuelOperations`](https://github.com/FuelLabs/fuels-ts/blob/mc/fix/ensure-correct-input-message-returned/packages/account/src/providers/transaction-summary/operations.ts#L201) as well which doesn't have access to an `assetId` to compare with the `baseAssetId` so I don't think that is possible.

Reply:
In this instance used in [`getWithdrawFromFuelOperations`](https://github.com/FuelLabs/fuels-ts/blob/mc/fix/ensure-correct-input-message-returned/packages/account/src/providers/transaction-summary/operations.ts#L201), could we not just pass through the `baseAssetId` as both arguments?

```ts
const input = getInputFromAssetId(inputs, baseAssetId, baseAssetId);
```

Reply:
I don't see how passing it that way offers better ergonomics or readability, passing the same parameter to a function twice is generally bad practice. I think the current signature conveys what's happening more clearly.

Reply:
@LuizAsFight, from my understanding, the helper `getInputFromAssetId` is designed to find the first resource input ([InputCoin](https://github.com/FuelLabs/fuels-ts/blob/4bc03879babf40df16a58c17e15b46093e519032/packages/transactions/src/coders/input.ts#L19) or [InputMessage](https://github.com/FuelLabs/fuels-ts/blob/4bc03879babf40df16a58c17e15b46093e519032/packages/transactions/src/coders/input.ts#L207)) based on a given asset ID.

Since all `InputMessage` are base asset ID resources, this helper can only return an `InputMessage` if the given asset ID is the base asset ID.

Therefore, without validation, there is no way to know if the given asset ID is the base asset ID, because an `InputMessage` does not have the property `assetId`, unlike `InputCoin`.

Please let me know if I am missing something here